lib/ansible/modules/cloud/heroku/heroku_collaborator.py

```diff
@@ -0,0 +1,142 @@
+    description:
+      - Suppress email invitation when creating collaborator
+    type: bool
+    required: false
+    default: "False"
+  user:
+    description:
+      - User ID or e-mail
+    required: true
+  state:
+    description:
+      - Create or remove the heroku collaborator
+    choices: ["present", "absent"]
+    required: false
+    default: "present"
+notes:
+  - C(HEROKU_API_KEY) and C(TF_VAR_HEROKU_API_KEY) env variable can be used instead setting c(api_key).
+  - If you use I(--check), you can also pass the I(-v) flag to see affected apps in msg.
+'''
+
+EXAMPLES = '''
+- heroku_collaborator:
+    api_key: YOUR_API_KEY
+    user: max.mustermann@example.com
+    apps: heroku-example-app
+    state: present
+
+- heroku_collaborator:
+    api_key: YOUR_API_KEY
+    user: '{{ item.user }}'
+    apps: '{{ item.apps | default(apps) }}'
+    suppress_invitation: '{{ item.suppress_invitation | default(suppress_invitation) }}'
+    state: '{{ item.state | default("present") }}'
+  with_items:
+    - { user: 'a.b@example.com' }
+    - { state: 'absent', user: 'b.c@example.com', suppress_invitation: false }
+    - { user: 'x.y@example.com', apps: ["heroku-example-app"] }
+'''
+
+RETURN = ''' # '''
+
+import os
+
+DEPENDENCY_CHECK = True
+try:
+    import heroku3
+except ImportError:
+    DEPENDENCY_CHECK = False
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def add_or_delete_heroku_collaborator(module, client):
+    user = module.params['user']
+    state = module.params['state']
+    affected_apps = []
+    result_state = False
+
+    for app in module.params['apps']:
+        if app not in client.apps():
+            module.fail_json(msg='App {0} does not exist'.format(app))
+
+        heroku_app = client.apps()[app]
+
+        heroku_collaborator_list = [collaborator.user.email for collaborator in heroku_app.collaborators()]
+
+        if state == 'absent' and user in heroku_collaborator_list:
+            if not module.check_mode:
+                heroku_app.remove_collaborator(user)
+            affected_apps += [app]
+            result_state = True
+        elif state == 'present' and user not in heroku_collaborator_list:
+            if not module.check_mode:
+                heroku_app.add_collaborator(user_id_or_email=user, silent=module.params['suppress_invitation'])
+            affected_apps += [app]
+            result_state = True
+
+    return result_state, affected_apps
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            api_key=dict(default=os.getenv('HEROKU_API_KEY', os.getenv('TF_VAR_HEROKU_API_KEY')), type='str'),
+            user=dict(required=True, type='str'),
+            apps=dict(required=True, type='list'),
+            suppress_invitation=dict(default=False, type='bool'),
+            state=dict(default='present', type='str', choices=['present', 'absent']),
+        ),
+        supports_check_mode=True
+    )
+
+    if not DEPENDENCY_CHECK:
```

Comment:
One suggestion, it will be helpful to create a common library under module_utils so that in future all Heroku module will use same like [DigitalOcean](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/digital_ocean.py) or [influxdb](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/influxdb.py).